### ISSUE TITLE
Class Cast Exception thrown from node transformer

### ISSUE THREAD

**Issue by nadeeshaan** [ballerina-platform/ballerina-lang#25910]
**Description:**
Consider the following source snippet,

```
import ballerina/java;

public type TypeDescName 

function foo(string filePath) {
    string [] columnHeaders = [];
    boolean testBool = true;
    if (testBool) {
        
    }
}

```

When tries to compile, this throws the following CCE,

```
Caused by: java.lang.ClassCastException: io.ballerinalang.compiler.syntax.tree.MappingBindingPatternNode cannot be cast to io.ballerinalang.compiler.syntax.tree.CaptureBindingPatternNode
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:545)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:435)
	at io.ballerinalang.compiler.syntax.tree.ModuleVariableDeclarationNode.apply(ModuleVariableDeclarationNode.java:67)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:530)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:435)
	at io.ballerinalang.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:460)
	at org.wso2.ballerinalang.compiler.parser.Parser.generateCompilationUnitNew(Parser.java:126)
	at org.wso2.ballerinalang.compiler.parser.Parser.parse(Parser.java:99)
	at org.wso2.ballerinalang.compiler.PackageLoader.parse(PackageLoader.java:487)
	at org.wso2.ballerinalang.compiler.PackageLoader.loadEntryPackage(PackageLoader.java:374)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:166)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:147)
	at org.ballerinalang.langserver.compiler.LSModuleCompiler.compilePackagesSafe(LSModuleCompiler.java:279)
	... 11 more
```

**Comment by lochana-chathura** [comment]
Simpler version: `student { };`

CCE should be gone away when implementing mapping binding pattern support in the compiler.

**Comment by rdulmina** [comment]
Fixed in #26949
